src/provider/store/notifications/index.ts

```diff
@@ -0,0 +1,4 @@
+import * as actions from './actions';
+import reducer from './reducer';
```

Comment:
I'm going to need a bit of a lesson on some of this. The contents of this directory all look to be doing something very similar to what we have in `APIHandler` and the `addListener` mechanics within the layouts/FDC3 clients, but with way more boilerplate than those two examples.

There's then also the `ui` folder with much of the same, and a corresponding `root-` file in the folder above. Perhaps this is partly down to redux conventions, but seems we could make this way more concise and still keep the type-safety?